Utils/azure-explorer-common/src/com/microsoft/tooling/msservices/serviceexplorer/azure/mysql/MySQLNode.java

```diff
@@ -0,0 +1,167 @@
+import com.microsoft.tooling.msservices.serviceexplorer.Node;
+import com.microsoft.tooling.msservices.serviceexplorer.NodeAction;
+import com.microsoft.tooling.msservices.serviceexplorer.NodeActionEvent;
+import com.microsoft.tooling.msservices.serviceexplorer.NodeActionListener;
+import com.microsoft.tooling.msservices.serviceexplorer.azure.AzureNodeActionPromptListener;
+import lombok.Getter;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.List;
+
+import static com.microsoft.tooling.msservices.serviceexplorer.azure.mysql.MySQLModule.ICON_FILE;
+
+
+public class MySQLNode extends Node {
+
+    @Getter
+    private final String subscriptionId;
+    @Getter
+    private final Server server;
+    private ServerState serverState;
+
+    public MySQLNode(AzureRefreshableNode parent, String subscriptionId, Server mysqlServer) {
+        super(mysqlServer.id(), mysqlServer.name(), parent, ICON_FILE, true);
+        this.subscriptionId = subscriptionId;
+        this.server = mysqlServer;
+        this.serverState = server.userVisibleState();
+        loadActions();
+    }
+
+    @Override
+    protected void loadActions() {
+        initActions();
+    }
+
+    @Override
+    public List<NodeAction> getNodeActions() {
+        boolean running = ServerState.READY.equals(serverState);
+        getNodeActionByName(StartAzureMySQLAction.ACTION_NAME).setEnabled(!running);
+        getNodeActionByName(StopAzureMySQLAction.ACTION_NAME).setEnabled(running);
+        getNodeActionByName(RestartAzureMySQLAction.ACTION_NAME).setEnabled(running);
+        return super.getNodeActions();
+    }
+
+    private void refreshNode() {
+        Server result = MySQLMvpModel.findServer(subscriptionId, server.resourceGroupName(), server.name());
+        this.serverState = result.userVisibleState();
+    }
+
+    // Delete Azure MySQL action class
+    @Name(DeleteAzureMySQLAction.ACTION_NAME)
+    public class DeleteAzureMySQLAction extends AzureNodeActionPromptListener {
+
+        private static final String ACTION_NAME = "Delete";
+        private static final String DELETE_PROMPT_MESSAGE =
+                "This operation will delete your azure mysql server: %s." + StringUtils.LF + "Are you sure you want to continue?";
+
+        public DeleteAzureMySQLAction() {
+            super(MySQLNode.this, String.format(DELETE_PROMPT_MESSAGE, MySQLNode.this.name),
+                    String.format(ACTION_NAME + MySQLModule.ACTION_PATTERN_SUFFIX, MySQLNode.this.name));
+        }
+
+        @Override
+        protected void azureNodeAction(NodeActionEvent e) {
+            MySQLNode.this.getParent().removeNode(MySQLNode.this.getSubscriptionId(), azureNode.getId(), azureNode);
+        }
+
+        @Override
+        protected void onSubscriptionsChanged(NodeActionEvent e) {
+
+        }
+    }
+
+    // Start Azure MySQL action class
+    @Name(StartAzureMySQLAction.ACTION_NAME)
+    public class StartAzureMySQLAction extends NodeActionListener {
```

Comment:
May create a new parent class for these actions to reduce duplicates later

Reply:
it is related to the separator logic. we can discuss this offline after separator group logic is ready.